### ISSUE TITLE
Fails if ARG is in a list separated by new lines, but not if they are ENVs

### ISSUE THREAD

**Issue by tinytyranid** [hadolint/hadolint#885]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior
Expected behaviour that it will recognise the ARGs in list separated by lines, as it does with ENVs

### Actual behavior
If arguments are given as ARGs in a list seperated by new lines hadolint will fail with the following warning: ` SC2154 warning: <some variable> is referenced but not assigned. `. But if the arguments are given as ENVs in a list separated by new lines there will be no warnings.

### Steps to reproduce the behavior
This Dockerfile will get `-:10 SC2154 warning: pangolin_ver is referenced but not assigned.`
```Docker
FROM ubuntu:20.04

ARG nextflow_ver="21.10.6" \
    miniconda_sh="Miniconda3-py38_4.12.0-Linux-x86_64.sh" \
    ncov_artic_ver="v1.3.0" \
    artic_ver="1.2.3" \
    primer_schemes_sha="9b533c7c29c3f08ec673aef6beced8ce98266267" \
    pangolin_ver="v4.1.2"

RUN echo "$nextflow_ver $miniconda_sh $ncov_artic_ver $artic_ver $primer_schemes_sha $pangolin_ver"
```

While this one will pass the hadolint:
```Docker
FROM ubuntu:20.04

ENV nextflow_ver="21.10.6" \
    miniconda_sh="Miniconda3-py38_4.12.0-Linux-x86_64.sh" \
    ncov_artic_ver="v1.3.0" \
    artic_ver="1.2.3" \
    primer_schemes_sha="9b533c7c29c3f08ec673aef6beced8ce98266267" \
    pangolin_ver="v4.1.2"

RUN echo "$nextflow_ver $miniconda_sh $ncov_artic_ver $artic_ver $primer_schemes_sha $pangolin_ver"
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

We have tested with the following versions of hadolint and both seem to produce the same warning using ARG, and passes using ENV:
```bash
v2.8.0
v2.9.2
```

**Additional environment details (OS, stack version, etc.)**
We used the official docker containers `hadolint/hadolint:v2.8.0-alpine` and `hadolint/hadolint:v2.9.2` for testing.

**Comment by m-ildefons** [comment]
Hey Camilla,

thanks for the bug report. While Docker allows specifying multiple arguments in a single `ARG` instruction, Podman does not[1]. I'm not sure what the correct behavior for a container runtime _should_ be in this case, but if you split up your `ARG` instruction, it works fine both in Docker and Podman and the warning goes away:

```Dockerfile
FROM ubuntu:20.04

ARG nextflow_ver="21.10.6"
ARG miniconda_sh="Miniconda3-py38_4.12.0-Linux-x86_64.sh"
ARG ncov_artic_ver="v1.3.0"
ARG artic_ver="1.2.3"
ARG primer_schemes_sha="9b533c7c29c3f08ec673aef6beced8ce98266267"
ARG pangolin_ver="v4.1.2"

RUN echo "$nextflow_ver $miniconda_sh $ncov_artic_ver $artic_ver $primer_schemes_sha $pangolin_ver"
```

Since `ARG` instructions don't mutate the overlayfs, this shouldn't impact the resulting image size at all.

Note that `ARG` and `ENV` aren't interchangeable for other reasons as well, because you can modify the value of a variable declared in an `ARG` instructions using the build-time `--build-arg VARIABLE=VALUE` option and they have different scopes than variables defined with `ENV`.

Interestingly, you can construct the Dockerfile using `ENV` with default values and `ARG` just to declare the variables:

```Dockerfile
FROM ubuntu:20.04

ARG nextflow_ver \
    miniconda_sh \
    ncov_artic_ver \
    artic_ver \
    primer_schemes_sha \
    pangolin_ver

ENV nextflow_ver=${nextflow_ver:-"21.10.6"} \
    miniconda_sh=${miniconda_sh:-"Miniconda3-py38_4.12.0-Linux-x86_64.sh"} \
    ncov_artic_ver=${ncov_artic_ver:-"v1.3.0"} \
    artic_ver=${artic_ver:-"1.2.3"} \
    primer_schemes_sha=${primer_schemes_sha:-"9b533c7c29c3f08ec673aef6beced8ce98266267"} \
    pangolin_ver=${pangolin_ver:-"v4.1.2"}

RUN echo "$nextflow_ver $miniconda_sh $ncov_artic_ver $artic_ver $primer_schemes_sha $pangolin_ver"
```
This won't build in Podman either, but works fine with Docker and Hadolint also doesn't complain.

@lorenzo I'm not sure if we should stick to the current behavior, which is consistent with Podman or support Docker's behavior here, but looking at the parser code, it doesn't seem right to just discard everything after the first argument:
```Haskell
parseArg :: (?esc :: Char) => Parser (Instruction Text)
parseArg = do
  reserved "ARG"
  (try nameWithDefault <?> "the arg name")
    <|> (try nameWithoutDefault <?> "the arg name")
    <|> Arg <$> untilEol "the argument name" <*> pure Nothing
  where
    nameWithoutDefault = do
      name <- someUnless "the argument name" (== '=')
      void $ untilEol "the rest"
      return $ Arg name Nothing
    nameWithDefault = do
      name <- someUnless "the argument name" (== '=')
      void $ char '='
      df <- untilEol "the argument value"
      return $ Arg name (Just df)
```
We probably shouldn't be using two completely different code paths for `ARG`s and `ENV`s anyways.

[1]: Podman throws this error: `Error: error building at STEP "ARG foo=bla bar=blabla": ARG requires exactly one argument definition`
